src/lib/expression/evaluation/expression_evaluator.cpp

```diff
@@ -0,0 +1,1054 @@
+
+    case ExpressionType::Select: {
+      const auto* pqp_select_expression = dynamic_cast<const PQPSelectExpression*>(&expression);
+      Assert(pqp_select_expression, "Can only evaluate PQPSelectExpression");
+      return _evaluate_select_expression<R>(*pqp_select_expression);
+    }
+
+    case ExpressionType::Column: {
+      const auto* pqp_column_expression = dynamic_cast<const PQPColumnExpression*>(&expression);
+      Assert(pqp_column_expression, "Can only evaluate PQPColumnExpressions");
+      return _evaluate_column_expression<R>(*pqp_column_expression);
+    }
+
+    // ValueExpression and ParameterExpression both need to unpack an AllTypeVariant, so one functions handles both
+    case ExpressionType::Parameter:
+    case ExpressionType::Value:
+      return _evaluate_value_or_parameter_expression<R>(expression);
+
+    case ExpressionType::Function:
+      return _evaluate_function_expression<R>(static_cast<const FunctionExpression&>(expression));
+
+    case ExpressionType::Case:
+      return _evaluate_case_expression<R>(static_cast<const CaseExpression&>(expression));
+
+    case ExpressionType::Exists:
+      return _evaluate_exists_expression<R>(static_cast<const ExistsExpression&>(expression));
+
+    case ExpressionType::Extract:
+      return _evaluate_extract_expression<R>(static_cast<const ExtractExpression&>(expression));
+
+    case ExpressionType::Negate:
+      return _evaluate_negate_expression<R>(static_cast<const NegateExpression&>(expression));
+
+    case ExpressionType::Aggregate:
+      Fail("ExpressionEvaluator doesn't support Aggregates, use the Aggregate Operator to compute them");
+
+    case ExpressionType::List:
+      Fail("Can't evaluate a ListExpression, lists should only appear as the right operand of an InExpression");
+  }
+}
+
+template <typename R>
+std::shared_ptr<ExpressionResult<R>> ExpressionEvaluator::_evaluate_arithmetic_expression(
+    const ArithmeticExpression& expression) {
+  const auto& left = *expression.left_operand();
+  const auto& right = *expression.right_operand();
+
+  // clang-format off
+  switch (expression.arithmetic_operator) {
+    case ArithmeticOperator::Addition:       return _evaluate_binary_with_default_null_logic<R, Addition>(left, right);
+    case ArithmeticOperator::Subtraction:    return _evaluate_binary_with_default_null_logic<R, Subtraction>(left, right);  // NOLINT
+    case ArithmeticOperator::Multiplication: return _evaluate_binary_with_default_null_logic<R, Multiplication>(left, right);  // NOLINT
+    case ArithmeticOperator::Division:       return _evaluate_binary_with_default_null_logic<R, Division>(left, right);
+
+    default:
+      Fail("ArithmeticOperator evaluation not yet implemented");
+  }
+  // clang-format on
+}
+template <>
+std::shared_ptr<ExpressionResult<int32_t>> ExpressionEvaluator::_evaluate_binary_predicate_expression<int32_t>(
```

Comment:
If we don't add `bool` to the list of supported types, let's at least have a central `using BoolSurrogate = int32_t` or something that is used in all places where we (ab)use ints for bools.

Reply:
yep, thought the same, let's discuss the best solution for this in person quickly